### PR TITLE
📖 Adjust README.md to fit v0.4.0 and bump k8s to v1.21.3

### LINKED ISSUE #943: v1alpha4: Update Readme
**Describe the solution you'd like**

With the upcoming CAPO release v0.4.0 we have to update the [README.md](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/README.md).
At least the compatibility matrix has to be adapted to the new version.

As the readme is part of the release (i.e., in the release archive), it should be done before we're releasing.

part of #745

/kind feature

### PR DESCRIPTION
**Pull request by tobiasgiese**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #943

Bumped Kubernetes version in the e2e tests to v1.21.2.
Further, adapted the tables format according to [the CAPI supported version in the CAPI books](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/book/src/reference/versions.md) 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**TODOs**:

- [x] squashed commits
- if necessary:
  - [ ] includes documentation
  - [ ] adds unit tests

/hold

<sub>Tobias Giese <tobias.giese@daimler.com>, Daimler TSS GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md)</sub>

### CODE REVIEW

**Review by chrischdi — COMMENTED** [review]
/ok-to-test

File: docs/book/src/clusteropenstack/configuration.md [hunk 1]
```diff
@@ -42,15 +42,15 @@ Note: You can use [the template file](../templates/cluster-template.yaml) by man
 # Using 'external-cloud-provider' flavor
 clusterctl config cluster capi-quickstart \
   --flavor external-cloud-provider \
-  --kubernetes-version v1.20.6 \
+  --kubernetes-version v1.21.2 \
```
Comment (chrischdi) [inline docs/book/src/clusteropenstack/configuration.md:45]: maybe we could consider using `v1.21.3`? (capi is still using v1.21.2 I think?)

Comment (tobiasgiese) [reply docs/book/src/clusteropenstack/configuration.md:45]: Correct, CAPI is using v1.21.2, that's why I decided to also use this version.
But okay for me to use the latest stable version

Comment (sbueringer) [reply docs/book/src/clusteropenstack/configuration.md:45]: I upgraded to 1.21.2 in CAPI because that was the latest release at that point. But in CAPI we also try to align the version we use for testing to our documentation. Not sure which version is used for testing in CAPO :)

Comment (sbueringer) [reply docs/book/src/clusteropenstack/configuration.md:45]: But the main CAPO quickstart doc is in the book anyway. So I guess it makes sense to align to the same version, but it doesn't really matter.

Comment (tobiasgiese) [reply docs/book/src/clusteropenstack/configuration.md:45]: @sbueringer the version for testing is also changed in this PR to v1.21.2 according to CAPI :)
I would suggest we not test a newer version as CAPI to avoid edge cases.
WDYT?

Comment (sbueringer) [reply docs/book/src/clusteropenstack/configuration.md:45]: Ups, overlooked that file.

It's fine to just use v1.21.2. It's pretty safe that v1.21.3 will work too. Imho it doesn't really matter which one you use

Comment (tobiasgiese) [reply docs/book/src/clusteropenstack/configuration.md:45]: Okay, if it really doesn't matter if we use a newer (patch) version than CAPI, then I'll change it to the current stable version

Comment (chrischdi) [reply docs/book/src/clusteropenstack/configuration.md]: :+1: for directly going to v1.21.3 . Hopefully we don't hit https://github.com/kubernetes/kubernetes/issues/102676 :-) but let's see (would also be the case for v1.21.2)

**Review by chrischdi — COMMENTED** [review]
/lgtm
